src/source-af-packet.c

```diff
@@ -782,6 +782,7 @@ int AFPReadFromRing(AFPThreadVars *ptv)
             SCReturnInt(AFP_FAILURE);
         }
         PKT_SET_SRC(p, PKT_SRC_WIRE);
+        p->pkt_mode = PKT_MODE_AUTO;
```

Comment:
I think the capture method has to tell the rest of the engine what mode it is in. So NFQ tells us we are inline, and AF_PACKET should tell us a packet is either IDS or IPS. It knows this, and the rest of the engine has no way of figuring this out without the help of the capture method.